geode-core/src/main/java/org/apache/geode/distributed/internal/InternalLocator.java

```diff
@@ -1108,6 +1108,7 @@ private boolean attemptReconnect() throws InterruptedException, IOException {
       InternalDistributedSystem newSystem =
           (InternalDistributedSystem) system.getReconnectedSystem();
       if (newSystem != null) {
+        setLocator(this);
```

Comment:
If locator restart fails this will leave the static _locator_ variable set.  Previously this wasn't the case.

Reply:
This was only case, if it was not set, in code above, when checking quorum. I will correct this.